packages/office-ui-fabric-react/src/components/ComboBox/ComboBox.tsx

```diff
@@ -1924,31 +1927,32 @@ export class ComboBox extends BaseComponent<IComboBoxProps, IComboBoxState> {
    * @returns No matter what specific types the input parameters are, always return an array of
    *  either strings or numbers instead of premitive type.  This normlization makes caller's logic easier.
    */
-  private _getSelectedKeys(
+  private _buildDefaultSelectedKeys(
     defaultSelectedKey: string | number | string[] | number[] | undefined,
     selectedKey: string | number | string[] | number[] | undefined
   ): string[] | number[] {
-    let retKeys: string[] | number[] = [];
-
-    if (defaultSelectedKey) {
-      if (defaultSelectedKey instanceof Array) {
-        retKeys = defaultSelectedKey;
-      } else if (typeof defaultSelectedKey === 'string') {
-        retKeys = [defaultSelectedKey as string];
-      } else if (typeof defaultSelectedKey === 'number') {
-        retKeys = [defaultSelectedKey as number];
-      }
-    } else if (selectedKey) {
-      if (selectedKey instanceof Array) {
-        retKeys = selectedKey;
-      } else if (typeof selectedKey === 'string') {
-        retKeys = [selectedKey as string];
-      } else if (typeof selectedKey === 'number') {
-        retKeys = [selectedKey as number];
-      }
+    const selectedKeys: string[] | number[] = this._buildSelectedKeys(defaultSelectedKey);
+    if (selectedKeys.length) {
+      return selectedKeys;
+    }
+
+    return this._buildSelectedKeys(selectedKey);
+  }
+
+  private _buildSelectedKeys(selectedKey: string | number | string[] | number[] | undefined): string[] | number[] {
+    if (typeof selectedKey === 'string' && selectedKey !== '') {
+      return [selectedKey];
+    }
+
+    if (typeof selectedKey === 'number' && selectedKey >= 0) {
```

Comment:
this is the weirdest bit of code I've seen. I don't have original context on this. @jspurlin do you ?

What is the intention here? It seems like it's don't nothing except filtering selectedKey values it probably shouldn't be!

Shouldn't this just be:

```
if (selectedKey === undefined) {
return [];
}

return (selectedKey instanceof Array ? selectedKey : [ selectedKey ]);
}
```

Reply:
(Also, I don't really understand why we'd special case blank strings. If the user passes in a blank string, they did something weird.)

Reply:
I think this code was added when multiselect comboBox functionality was added

---
In reply to: [205835697](https://github.com/OfficeDev/office-ui-fabric-react/pull/5705#discussion_r205835697) [](ancestors = 205835697)

Reply:
@jspurlin thank you very much for the clarification 👍 :)

Reply:
@dzearing I think, your suggestion will cause the type error, [Example, need to enable `strictNullChecks`](https://www.typescriptlang.org/play/#src=class%20ComboBox%20%7B%0D%0A%20%20%20%20private%20_buildSelectedKeys(selectedKey%3A%20string%20%7C%20number%20%7C%20string%5B%5D%20%7C%20number%5B%5D%20%7C%20undefined)%3A%20string%5B%5D%20%7C%20number%5B%5D%20%7B%0D%0A%20%20%20%20%20%20%20%20if%20(typeof%20selectedKey%20%3D%3D%3D%20'string')%20%7B%0D%0A%20%20%20%20%20%20%20%20%20%20%20%20return%20%5B%5D%3B%0D%0A%20%20%20%20%20%20%20%20%7D%0D%0A%0D%0A%20%20%20%20%20%20%20%20return%20(selectedKey%20instanceof%20Array%20%3F%20selectedKey%20%3A%20%5BselectedKey%5D)%3B%0D%0A%20%20%20%20%7D%0D%0A%7D)

Maybe that's why the previous version had code dublication

```
let retKeys: string[] | number[] = [];

if (defaultSelectedKey) {
  if (defaultSelectedKey instanceof Array) {
    retKeys = defaultSelectedKey;
  } else if (typeof defaultSelectedKey === 'string') {
    retKeys = [defaultSelectedKey as string];
  } else if (typeof defaultSelectedKey === 'number') {
    retKeys = [defaultSelectedKey as number];
  }
} else if (selectedKey) {
  if (selectedKey instanceof Array) {
    retKeys = selectedKey;
  } else if (typeof selectedKey === 'string') {
    retKeys = [selectedKey as string];
  } else if (typeof selectedKey === 'number') {
    retKeys = [selectedKey as number];
  }
}

return retKeys;
```